### PR TITLE
Makefile improvements (install man pages) #35

### PR DESCRIPTION
**Pull request by csirac2**
Breaking change: INSTALL_PREFIX should now be a path to a directory containing bin & share/man/man8 directories, rather than a full path to the bin directory itself, like how the rest of the world works.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,21 +1,46 @@
...
 
-test: bats-tests prune-tests
+$(INSTALL_PREFIX)/bin/%: %
+	install -Dm755 $< $@
+
+install-man: $(addprefix $(INSTALL_PREFIX)/share/man/man8/, $(addsuffix .8.gz, \
+	$(call ls_bin)))
+
+$(INSTALL_PREFIX)/share/man/man8/%.8.gz: man/%.8.gz | \
+	$(INSTALL_PREFIX)/share/man/man8
```
Comment (florianjacob) [inline Makefile]: This is redundant, the -D option of install creates all leading directories.

File: Makefile [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
+	install -Dm644 $< $@
+
+$(INSTALL_PREFIX)/share/man/man8:
+	mkdir $(INSTALL_PREFIX)/share/man/man8
```
Comment (florianjacob) [inline Makefile]: This line broke in my tests, as I had no `/usr/local/share/man` directory. Initially I would have gone with a `mkdir -p`, but I'm quite sure this target can actually be left out; see other comment.

File: Makefile (hunk 1 above)
Comment (florianjacob) [reply Makefile]: Removing the additional target completely has resolved all my issues. :smile:

Slightly Off Topic: What does the `|` in Makefile target specifications like the line above actually do? This is really hard to search for with a search engine…

**Review by florianjacob — COMMENTED** [review]
Looks good, apart from the minor fix required in directory creation which will even simplify the code a little.

File: Makefile (hunk 1 above)
Comment (csirac2) [reply Makefile]: Didn't know about -D, I learn something every day! Things after the | are "order-only" prerequisites; that basically boils down to, "satisfy this dependency if it's missing, but don't bother rebuilding the target if it changes"

**Comment by csirac2** [comment]
Rebased & removed the redundant mkdir stuff. Will let you merge once you're happy @jhass is ready.

File: Makefile (hunk 1 above)
Comment (florianjacob) [reply Makefile]: Thanks, makes sense. :+1:

**Comment by florianjacob** [comment]
I'm happy now. :smile: I reached out to jhass, let's see what he thinks about this. I will merge as soon as he's happy as well.

**Pull request merged**